### PR TITLE
Added section about linters

### PR DESCRIPTION
**Pull request by esteve**

### CODE REVIEW

**Comment by esteve** [comment]
This is still pending confirmation from upstream in https://github.com/ament/ament_lint/pull/278, do not merge yet.

File: docs/porting-to-ROS2.md [hunk 1]
```diff
@@ -111,6 +111,24 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 ```
 
+#### Linters
+Add only `ament_cmake_cppcheck` and `ament_cmake_cpplint` to the list of linters in `package.xml`
+
+```xml
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_cmake_cppcheck</test_depend>
+  <test_depend>ament_cmake_cpplint</test_depend>
```
Comment (jilaada) [inline docs/porting-to-ROS2.md]: Maybe I missed some discussion but I thought that we would just be doing cppcheck and clang-tidy?

File: docs/porting-to-ROS2.md [hunk 2]
```diff
@@ -111,6 +111,24 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
...
+
+And the corresponding code in `CMakeLists.txt`
+
+```cmake
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+endif()
+```
+
```
Comment (jilaada) [inline docs/porting-to-ROS2.md:130]: Might be nice to add here that in order to run the run the tests and see the output you can run `colcon test --packages-select <pkg_name> && colcon test-result --verbose`. In addition, to get clang-tidy to work you need to add the cmake flags: `--cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=1`

**Review by jilaada — COMMENTED** [review]
Just a couple of comments

File: docs/porting-to-ROS2.md (hunk 1 above)
Comment (esteve) [reply docs/porting-to-ROS2.md]: I must have misheard @dejanpan, I thought he said he, @nnmm and @fred-apex-ai agreed to use cpplint as well.

Comment (esteve) [reply docs/porting-to-ROS2.md]: Fixed in e7a8d65

File: docs/porting-to-ROS2.md (hunk 2 above)
Comment (esteve) [reply docs/porting-to-ROS2.md:130]: Added in dfc2733

File: docs/porting-to-ROS2.md (hunk 1 above)
Comment (jilaada) [reply docs/porting-to-ROS2.md]: Thanks for clarifying!

**Review by jilaada — APPROVED** [review]
Changes look good to me. Thanks for adding this :+1:

**Pull request merged**